area/guadosalam.py

```diff
@@ -15,113 +16,74 @@
 
 def arrival():
     logger.info("Starting Guadosalam section")
+    checkpoint=0
     memory.main.click_to_control()
-
-    FFXC.set_movement(-1, 1)
-    memory.main.wait_frames(30 * 0.5)
-    FFXC.set_movement(0, 1)
-    memory.main.wait_frames(30 * 3.5)
-    FFXC.set_movement(1, 1)
-    memory.main.wait_frames(30 * 0.2)
-    FFXC.set_movement(0, 1)
-    memory.main.wait_frames(30 * 0.6)
-    FFXC.set_neutral()
-
-    memory.main.click_to_control_3()
-    FFXC.set_movement(0, -1)
-    memory.main.wait_frames(30 * 1)
-    FFXC.set_neutral()
-
-    memory.main.click_to_control_3()
-    FFXC.set_movement(0, 1)
-    memory.main.wait_frames(30 * 2)
-    FFXC.set_neutral()  # Enter the room where we meet Seymour
-
-    logger.debug(f"Test Var (CSR) - {game_vars.csr}")
-    # Adjusted branch CSR logic, start
-    memory.main.click_to_control_3()
-    if game_vars.csr():
-        while not pathing.set_movement([-13, -67]):
-            pass
-        logger.debug("Lulu conversation")
-        while memory.main.user_control():  # Lulu conversation
-            pathing.set_movement([-11, -55])
-            xbox.tap_b()
-        FFXC.set_neutral()
-        memory.main.click_to_control_3()
-
-        while not pathing.set_movement([-39, -77]):
-            pass
-        logger.debug("Wakka conversation")
-        while memory.main.user_control():  # Start conversation with Wakka
-            pathing.set_movement([-49, -61])
-            xbox.tap_b()
-        FFXC.set_neutral()
-        memory.main.click_to_control_3()
-
-        while not pathing.set_movement([4, -114]):
-            pass
-        logger.debug("Talk to Auron")
-        while memory.main.user_control():  # Talk to Auron
-            pathing.set_movement([18, -119])
-            xbox.tap_b()
-        FFXC.set_neutral()
-        memory.main.click_to_control_3()
-
-    else:
-        while not pathing.set_movement([4, -114]):
-            pass
-        logger.debug("Talk to Auron")
-        while memory.main.user_control():  # Talk to Auron (first for affection)
-            pathing.set_movement([18, -119])
-            xbox.tap_b()
-        FFXC.set_neutral()
-        memory.main.click_to_control_3()
-
-        while not pathing.set_movement([-39, -77]):
-            pass
-        logger.debug("Wakka conversation")
-        while memory.main.user_control():  # Start conversation with Wakka
-            pathing.set_movement([-49, -61])
-            xbox.tap_b()
-        FFXC.set_neutral()
-        memory.main.click_to_control_3()
-
-        while not pathing.set_movement([-13, -67]):
-            pass
-        logger.debug("Lulu conversation")
-        while memory.main.user_control():  # Lulu conversation
-            pathing.set_movement([-11, -55])
+    while memory.main.get_map() != 141: #Up to the dining hall scenes
+        if memory.main.user_control():
+            if checkpoint == 4:
```

Comment:
Some day, I'd like to update the pathing classes to handle stuff like this. But that's a pretty significant change. No reason to do that here. Just dropping the comment so it's not only in my head.